src/coreclr/jit/importer.cpp

```diff
@@ -5486,7 +5486,8 @@ GenTree* Compiler::impCastClassOrIsInstToTree(
         else if (isCastClass)
         {
             // Jit can only inline expand CHKCASTCLASS and CHKCASTARRAY helpers.
-            canExpandInline = (helper == CORINFO_HELP_CHKCASTCLASS) || (helper == CORINFO_HELP_CHKCASTARRAY);
+            canExpandInline = (helper == CORINFO_HELP_CHKCASTCLASS) || (helper == CORINFO_HELP_CHKCASTARRAY) ||
+                              (helper == CORINFO_HELP_CHKCASTANY);
         }
         else if ((helper == CORINFO_HELP_ISINSTANCEOFCLASS) || (helper == CORINFO_HELP_ISINSTANCEOFARRAY))
```

Comment:
Should this be done for isinst too?

Reply:
Probably? The Bing's stats for `IsInstanceOfAny` are:
```
ttotal=25578135, iisNull=2278487, iisSame=3925180

15% same
10% is null
```

Reply:
I didn't add `IsInstanceOfAny` because we don't oportunistically expand it even for non-generics today, e.g.
```csharp
bool isinst = obj is MyClass1;
```
(we only do that when `MyClass1` is sealed and won't even emit a fallback)

we only do that for castclass, e.g. `(MyClass1)obj`. So presumably I'll have to slightly alter the logic to support `IsInstanceOfAny` + fallback